src/main/java/com/javadiscord/javabot/StatsUpdater.java

```diff
@@ -0,0 +1,44 @@
+package com.javadiscord.javabot;
+
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.OnlineStatus;
+import net.dv8tion.jda.api.events.ReadyEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class StatsUpdater extends ListenerAdapter {
+
+    private final ScheduledExecutorService threadPool = Executors.newSingleThreadScheduledExecutor();
+
+    private final long delay = 20;
+
+    private final TimeUnit delayUnit = TimeUnit.MINUTES;
+
+    private JDA jda;
```

Comment:
I don't think there's any reason to keep a reference to JDA here, since you only ever use it in the onReady event where it's originally obtained.